src/ui/starred-files-tab.h

```diff
@@ -48,7 +49,7 @@ private slots:
     QWidget *loading_failed_view_;
     QWidget *empty_view_;
 
-    GetStarredFilesRequest *get_starred_files_req_;
+    QScopedPointer<GetStarredFilesRequest> get_starred_files_req_;
```

Comment:
Ehh, `QScopedPointer` (aka something like `std::unique_ptr`) is simple and useful to keep the lifetime of the object allocated from heap, because you know how it is managed and when it is destroyed (it is destroyed only the wrapper object is destroyed). The way you mentioned (deleting itself after completed) is regaining the memory faster. However, in the later way, it is much more complicate that you need to add additional slot to delete itself but you can hardly tell if others slots(or handlers) require the access of this object. So it can be tricky to make it memory-safe and correct.

In fact, I am good to use more code for more efficient stuff and if you want to do it, `deleterLater` instead of `operator delete` might help. If you want to keep in a simple way, it is good enough now..